src/OpenTelemetry.ResourceDetectors.Host/HostDetector.cs

```diff
@@ -33,4 +79,88 @@ public Resource Detect()
 
         return Resource.Empty;
     }
+
+    private static IEnumerable<string> GetFilePaths()
+    {
+        yield return ETCMACHINEID;
+        yield return ETCVARDBUSMACHINEID;
+    }
+
+    private static string? GetMachineIdMacOs()
+    {
+        try
+        {
+            var startInfo = new ProcessStartInfo
+            {
+                FileName = "sh",
+                Arguments = "ioreg -rd1 -c IOPlatformExpertDevice | awk '/IOPlatformUUID/ { split($0, line, \"\\\"\"); printf(\"%s\\n\", line[4]); }'",
```

Comment:
I'd prefer to have processing of results of `ioreg -rd1 -c IOPlatformExpertDevice` be done in-proc